File: relayboard/logtooth/src/main/java/pers/hal42/android/Logger.java
package pers.hal42.android;

import android.util.Log;

/**
 * Copyright (C) by andyh created on 1/9/13 at 10:57 AM
 * convenience wrapper on android Log usage, not having to type a tag at each place of use.
 * Also serves as a nice place to set breakpoints to trap otherwise discarded exception messages.
 */
public class Logger {
  public final String logTag;

  public Logger(String tag) {
    logTag = tag;
  }

  public void i(String format, Object... args) {
    Log.i(logTag, String.format(format, args));
  }

  public void d(String format, Object... args) {
    Log.d(logTag, String.format(format, args));
  }

  public void e(String format, Object... args) {
    Log.e(logTag, String.format(format, args));
  }

}

File: android/src/main/kotlin/pers/hal42/android/Logger.kt
package pers.hal42.android

import android.util.Log

/**
 * Copyright (C) by andyh created on 1/9/13 at 10:57 AM
 * convenience wrapper on android Log usage, not having to type a tag at each place of use.
 * Also serves as a nice place to set breakpoints to trap otherwise discarded exception messages.
 */
class Logger(val logTag: String?) {

  fun i(format: String, vararg args: Any) {
    Log.i(logTag?:"?", String.format(format, *args))
  }

  fun d(format: String, vararg args: Any) {
    Log.d(logTag?:"?", String.format(format, *args))
  }

  fun e(format: String, vararg args: Any) {
    Log.e(logTag?:"?", String.format(format, *args))
  }

}

val Dbg=Logger("Application")